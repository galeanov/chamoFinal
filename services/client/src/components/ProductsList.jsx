import React from 'react';


const ProductsList = (props) => {
  return (
    <div>
      <table className="table is-narrow is-striped is-fullwidth is-hoverable">
      <thead>
        <tr>
          <th>Nombre</th>
          <th>Precio</th>
          <th>Cantidad</th>
          <th>Descripcion</th>
          <th>Categoria</th>
        </tr>
      </thead>
      <tbody>
      {
        props.products.map((product) => {
          return (
            <tr key={product.id}>
            <td>{ product.nombre}</td>
            <td>{ product.precio}</td>
            <td>{ product.cantidad}</td>
            <td>{ product.descripcion}</td>
            <td>{ product.categoria}</td>
            </tr>
          )
        })
      }
      </tbody>
      </table>
    </div>
  )
};


export default ProductsList;