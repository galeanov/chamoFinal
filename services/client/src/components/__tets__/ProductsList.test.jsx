import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
 
import ProductsList from '../ProductsList';
 
const products = [
  {
    'active': true,
    'cantidad': 12,
    'precio': 0.12,
    'descripcion': 'esta rico',
    'categoria': 'Galletas',
    'id': 1,
    'nombre': 'soda'
  },
  {
    'active': true,
    'cantidad': 20,
    'precio': 0.7,
    'descripcion': 'esta buenaso',
    'categoria': 'Galletas',
    'id': 2,
    'nombre': 'soda field'
  }
];
 
test('ProductsList renders properly', () => {
  const wrapper = shallow(<ProductsList products={products}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('Oreo');
});

test('ProductsList renders a snapshot properly', () => {
  const tree = renderer.create(<ProductsList products={products}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
