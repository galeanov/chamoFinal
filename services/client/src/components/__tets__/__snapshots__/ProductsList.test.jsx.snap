// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList renders a snapshot properly 1`] = `
<div>
  <table
    className="table is-narrow is-striped is-fullwidth is-hoverable"
  >
    <thead>
      <tr>
        <th>
          Nombre
        </th>
        <th>
          Precio
        </th>
        <th>
          Cantidad
        </th>
        <th>
          Descripcion
        </th>
        <th>
          Categoria
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          soda
        </td>
        <td>
          0.12
        </td>
        <td>
          12
        </td>
        <td>
          esta rico
        </td>
        <td>
          Galletas
        </td>
      </tr>
      <tr>
        <td>
          soda field
        </td>
        <td>
          0.7
        </td>
        <td>
          20
        </td>
        <td>
          esta buenaso
        </td>
        <td>
          Galletas
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
